internal/git/git.go

```
@@ -101,67 +100,55 @@ type BranchShas struct {
-		if err != nil {
-			logger.Error(err, "could not git checkout", "gitError", stderr)
-			return nil, err
-		}
-		logger.V(4).Info("Checked out branch", "branch", branch)
+	_, stderr, err = g.runCmd(ctx, g.pathLookup.Get(g.gap.GetGitHttpsRepoUrl(*g.gitRepo)+g.pathContext), "pull", "--progress")
+	if err != nil {
+		logger.Error(err, "could not git pull", "gitError", stderr)
+		return BranchShas{}, err
+	}
+	logger.V(4).Info("Pulled branch", "branch", branch)
 
-		_, stderr, err = g.runCmd(ctx, g.pathLookup.Get(g.gap.GetGitHttpsRepoUrl(*g.gitRepo)+g.pathContext), "pull", "--progress")
-		if err != nil {
-			logger.Error(err, "could not git pull", "gitError", stderr)
-			return nil, err
-		}
-		logger.V(4).Info("Pulled branch", "branch", branch)
+	stdout, stderr, err := g.runCmd(ctx, g.pathLookup.Get(g.gap.GetGitHttpsRepoUrl(*g.gitRepo)+g.pathContext), "rev-parse", branch)
+	if err != nil {
+		logger.Error(err, "could not get branch shas", "gitError", stderr)
+		return BranchShas{}, err
+	}
 
-		stdout, stderr, err := g.runCmd(ctx, g.pathLookup.Get(g.gap.GetGitHttpsRepoUrl(*g.gitRepo)+g.pathContext), "rev-parse", branch)
-		if err != nil {
-			logger.Error(err, "could not get branch shas", "gitError", stderr)
-			return nil, err
-		}
+	shas := BranchShas{}
+	shas.Hydrated = strings.TrimSpace(stdout)
+	logger.V(4).Info("Got hydrated branch sha", "branch", branch, "sha", shas.Hydrated)
 
-		if _, ok := shaMap[branch]; !ok {
-			shaMap[branch] = &BranchShas{}
-		}
-		shaMap[branch].Hydrated = strings.TrimSpace(stdout)
-		logger.V(4).Info("Got hydrated branch sha", "branch", branch, "sha", shaMap[branch].Hydrated)
-
-		// TODO: safe path join
-		metadataFile := g.pathLookup.Get(g.gap.GetGitHttpsRepoUrl(*g.gitRepo)+g.pathContext) + "/hydrator.metadata"
-		if _, err := os.Stat(metadataFile); err != nil {
```

I removed this initial stat check, os.Open should throw the same error.